src/main/java/org/apache/flink/benchmark/KeyByBenchmarks.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.benchmark;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.streaming.api.environment.LocalStreamEnvironment;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.DiscardingSink;
+import org.apache.flink.streaming.api.functions.source.ParallelSourceFunction;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+/**
+ * Benchmark for keyBy() on tuples and arrays.
+ */
+public class KeyByBenchmarks extends BenchmarkBase {
+
+	private static final int TUPLE_RECORDS_PER_INVOCATION = 25_000_000;
```

Comment:
Does those numbers of records must be that high? Other benchmarks are using `7_000_000`.

The lower the number of records, the faster or the more accurate the benchmarks will be. JMH executes as many invocations in a single run as possible as long as in total they do not exceed 1 second. Thus:
- if one invocation is already > 1 sec, smaller number of records => faster benchmarks
- if one invocation is already < 1 sec, smaller number of records => more invocations => more accurate results.

The only reason for larger number of records is if profiling shows that initialising/rump up is taking too much time.

Reply:
Indeed, fewer records also resulted in the same values. For this simple job, however, the ramp-up is a significant portion of the benchmark if the number of records it too small.